client/extensions/woocommerce/app/settings/shipping/shipping-zone/index.js

```diff
@@ -49,18 +53,35 @@ class Shipping extends Component {
 	}
 
 	render() {
-		const { siteId, className, loaded, markSaved, markChanged, params } = this.props;
+		const { siteId, className, loaded, markSaved, markChanged, params, zone, locations, actions, translate } = this.props;
 		const isRestOfTheWorld = 0 === Number( params.zone );
 
+		const onSave = () => {
+			actions.changeShippingZoneName( siteId, getZoneName( zone, locations, translate ) );
```

Comment:
Which, looking at your code, is seems that's where you have drawn inspiration from 😂